addons/AutoExportVersion/AutoExportVersion.gd

```diff
@@ -1,147 +1,185 @@
-#		else:
-#			# Edit formatting here.
-#			return "%s %s" % [code, vname]
-#	else:
-#		push_error("Failed to fetch version. export_presets.cfg does not exist.")
+	for section in config.get_sections():
+		if not section.ends_with(".options"):
+			continue
+		for key in config.get_section_keys(section):
+			for check_key in version_keys:
+				if key.ends_with(check_key):
+					var version: String =  str(config.get_value(section, key))
+					if version.is_empty():
+						continue 
+					return version
 	
+	push_error("Failed to fetch version. No valid version key found in export profiles.")
 	return ""
 
-### Unimportant stuff here.
+## Stores a [param version] based on [param version_store_location].                            [br]
+## See [member PROJECT_SETTING_NAME], [member SCRIPT_PATH]
+func store_version(version: String, version_store_location: VersionStoreLocation=VersionStoreLocation.PROJECT_SETTING) -> void:
+	if version.is_empty():
+		printerr("Cannot store version. " + _EMPTY_VERSION_ERROR.format({"script_path": get_script().get_path()}))
+		return
+	
+	match version_store_location:
+		VersionStoreLocation.SCRIPT:
+			store_version_as_script(version)
+		VersionStoreLocation.PROJECT_SETTING:
+			store_version_as_project_setting(version)
+
+## Stores the version in ProjectSettings.                                                       [br]
+## If the [param persistent] is true, then it is going to be written to the project.godot as well.
+func store_version_as_project_setting(version: String, persistent: bool=true) -> void:
+	ProjectSettings.set_setting(PROJECT_SETTING_NAME, version)
+	if persistent:
+		ProjectSettings.save()
+		ProjectSettings.set_initial_value(PROJECT_SETTING_NAME, "Empty version")
+		ProjectSettings.add_property_info({
+			"name": PROJECT_SETTING_NAME,
+			"type": TYPE_STRING,
+			"hint": PROPERTY_HINT_PLACEHOLDER_TEXT,
+			"hint_string": "Will overriden on export by AutoExportVersion plugin"
+		})
+
+## Stores the version as a script based on [member SCRIPT_TEMPLATE] in [member SCRIPT_PATH].
+func store_version_as_script(version: String) -> void:
+	var script: GDScript = GDScript.new()
+	script.source_code = SCRIPT_TEMPLATE.format({"version": version})
+	var err: int = ResourceSaver.save(script, SCRIPT_PATH)
+	if err:
+		push_error("Failed to save version as script. Error: %s" % error_string(err))
+
+
+
+####################################################################################################
+#################################  Unimportant plugin stuff here  ##################################
+####################################################################################################
+
+
+const _CURRENT_VERSION: String = "Current version: {version}"
+const _EMPTY_VERSION_ERROR: String = "Version string is empty.\nMake sure your 'get_version()' in '{script_path}' is configured properly."
+
+const _TOOL_MENU_ITEM_NAME: String = "AutoExport: Print Current Version"
```

Comment:
IMO adding constants for stuff that is used once is rather redundant. But keeping it is ok I guess.

Reply:
I wasn't a fan of those super long lines so this was my way to shorten them while making it easier to modify the value without breaking anything or making inconsistencies.

I am not 100% sure about this myself